app/src/form/__test__/FormContainer.test.js

```diff
@@ -276,36 +286,53 @@ describe('FormContainer component on submit', () => {
       7: ['Fried Rice'],
       8: ' ',
     };
+    act(() => {
+      FormState.update((s) => {
+        s.currentValues = modifiedInitialValues;
+      });
+    });
 
-    const { queryByTestId } = render(
+    const { queryByTestId, rerender } = render(
       <FormContainer
         forms={exampleTestForm}
         initialValues={modifiedInitialValues}
         onSubmit={handleOnSubmit}
       />,
     );
-
     const formSubmitBtn = queryByTestId('form-btn-submit');
     expect(formSubmitBtn).toBeDefined();
-    fireEvent.press(formSubmitBtn);
 
-    await waitFor(() => expect(handleOnSubmit).toHaveBeenCalledTimes(1));
-    expect(handleOnSubmit).toHaveBeenCalledWith(
-      {
-        name: 'John',
-        geo: null,
-        answers: {
-          1: 'John',
-          2: new Date('01-01-1992'),
-          3: '31',
-          4: ['Male'],
-          5: ['Bachelor'],
-          6: ['Traveling'],
-          7: ['Fried Rice'],
-        },
-      },
-      expect.any(Function),
+    act(() => {
+      fireEvent.press(formSubmitBtn);
+    });
+
+    rerender(
+      <FormContainer
+        forms={exampleTestForm}
+        initialValues={modifiedInitialValues}
+        onSubmit={handleOnSubmit}
+      />,
     );
+
+    await waitFor(() => {
+      //        expect(handleOnSubmit).toHaveBeenCalledTimes(1)
```

Comment:
@ifirmawan why you commented this?

Reply:
oh wait mas I think I missed the undo after debugging